### PR TITLE
Generate and remove files with system user

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1042

### Change description ###
Generate and remove files using the system user
Remove requirement for null argument when creating filename

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/CreateGeneralOrder.java [hunk 1]
```diff
@@ -37,24 +35,21 @@
...
-                .mandatory(GeneralOrder::getGeneralOrderDetails)
-                .done();
+            .mandatory(GeneralOrder::getGeneralOrderDate)
+            .mandatory(GeneralOrder::getGeneralOrderDivorceParties)
+            .optional(GeneralOrder::getGeneralOrderRecitals)
+            .mandatory(GeneralOrder::getGeneralOrderJudgeType)
+            .mandatory(GeneralOrder::getGeneralOrderJudgeName)
+            .mandatory(GeneralOrder::getGeneralOrderLegalAdvisorName)
+            .mandatory(GeneralOrder::getGeneralOrderDetails)
+            .done();
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/CreateGeneralOrder.java:52]: formatting :)

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java [hunk 2]
```diff
@@ -49,17 +44,16 @@
 
             log.info("Generating citizen respondent AoS invitation for case id {} ", caseId);
             final Supplier<Map<String, Object>> templateContentSupplier = templateContent.apply(caseData, caseId, createdDate);
-            final String userAuthorisation = request.getHeader(AUTHORIZATION);
+            final Supplier<String> filename = () -> RESP_AOS_INVITATION_DOCUMENT_NAME + caseId;
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java:47]: any reason filename is passed as a supplier ?

Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java]: Removed supplier 
https://github.com/hmcts/nfdiv-case-api/pull/582/commits/3d69fcef3e2f6698044af988dc393febb7548193

**Review by nitinprabhu — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/CreateGeneralOrder.java (hunk 1 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/CreateGeneralOrder.java:52]: looks like IDE again formatted it :)

Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/CreateGeneralOrder.java]: fixed again 👍

**Pull request merged**